### DISCUSSION TITLE
Infinite redirect on 404 not found page (production)

### DISCUSSION THREAD

**Discussion by Aerilym** [vercel/next.js#59142] [Help] [answered]
### Summary

When navigating to a non-existent path on my website the custom `not-found.tsx` loads but the site seems to infinitely refresh. This appears similar to the following:
#50732
#10024
#50429

These other issues seem to only apply to development mode, but this issue is occurring on my production site.
1. Is it possible that the site is running in development mode? and if so, how can I change that.
2. Have I misconfigured the not-found page?

Note: the not-found.tsx file is the only one in the project and is in the /app directory.

Thanks in advance for any assistance.

Production website not found link: https://katapod.com/this-page-does-not-exist

## Not found file
```tsx
// app/not-found.tsx
import { Button } from '@katapod/ui/components/ui/button';

export default function NotFound() {
  return (
    <div className="relative isolate px-6 pt-10 lg:px-8">
      <div className="mx-auto max-w-2xl py-16 sm:py-36 lg:py-44">
        <div className="text-center">
          <h1 className="text-4xl font-bold tracking-tight text-gray-900 sm:text-6xl">404</h1>
          <p className="mt-6 text-lg leading-8 text-gray-600">Page not found!</p>

          <div className="mt-10 flex items-center justify-center gap-x-6">
            <a href="#">
              <Button size="lg" variant="secondary">
                Return Home
              </Button>
            </a>
          </div>
        </div>
      </div>
      <div
        className="absolute inset-x-0 top-[calc(100%-13rem)] -z-10 transform-gpu overflow-hidden blur-3xl sm:top-[calc(100%-30rem)]"
        aria-hidden="true"
      >
        <div
          className="relative left-[calc(50%+3rem)] aspect-[1155/678] w-[36.125rem] -translate-x-1/2 bg-gradient-to-tr from-[#ff80b5] to-[#9089fc] opacity-30 sm:left-[calc(50%+36rem)] sm:w-[72.1875rem]"
          style={{
            clipPath:
              'polygon(74.1% 44.1%, 100% 61.6%, 97.5% 26.9%, 85.5% 0.1%, 80.7% 2%, 72.5% 32.5%, 60.2% 62.4%, 52.4% 68.1%, 47.5% 58.3%, 45.2% 34.5%, 27.5% 76.7%, 0.1% 64.9%, 17.9% 100%, 27.6% 76.8%, 76.1% 97.7%, 74.1% 44.1%)',
          }}
        />
      </div>
    </div>
  );
}
```

### Additional information

![image](https://github.com/vercel/next.js/assets/5667907/ffcab148-6293-4dae-b17d-ec7da96d10d9)

Versions
next-js: 14.0.2
react: 18.2.0
react-dom: 18.2.0
node: 18.18.1

### Example

https://katapod.com/this-page-does-not-exist

**Comment by aziyatali** [comment]
@Aerilym  Can you please share your page structure inside app folder? I want to exactly see the not-found.js page.

**Reply by Aerilym** [reply]
![image](https://github.com/vercel/next.js/assets/5667907/1f697370-840b-4c17-b11c-6334063bd0ec)

**Reply by Aerilym** [reply]
The not-found page is there, you can see it at the link I added. https://katapod.com/this-page-does-not-exist
But it infinitely reloads.

**Reply by aziyatali** [reply]
Check, your have both page.tsx and not-found.tsx in the same layout. Separate them and try.

**Reply by Aerilym** [reply]
How do I check they are in the same layout? Doesn't their position in the app directory make them children of `layout.tsx`?

**Reply by aziyatali** [reply]
Yes, they are children of layout.tsx but you have both page.tsx and not-found.tsx at the same level and they are same thing which is used for navigation. Check app routing of NextJs.

**Reply by Aerilym** [reply]
Sorry I don't understand how that is related to the issue? The not-found page is infinitely refreshing regardless of what page level triggers it (and if manually triggered). see the link I sent. You can also try navigating to others like /login/this-page-is-not-here and it does the same.

> In addition to catching expected notFound() errors, the root app/not-found.js file also handles any unmatched URLs for your whole application. This means users that visit a URL that is not handled by your app will be shown the UI exported by the app/not-found.js file.
https://nextjs.org/docs/app/api-reference/file-conventions/not-found

**Reply by aziyatali** [reply]
Ah, my bad. I thought you have multiple layouts. I am using Next 13.5 version and it is working quite fine. 
Would you mind sharing your Nextjs version and others infos that might be useful for community to figure out.

**Reply by Aerilym** [reply]
All good, yeah ill add the details to the post.

**Comment by 2010363329** [comment]
贴个照片看看  目测打错单词了

**Reply by Aerilym** [reply]
Sure, added it. To be clear, the not found page is there, you can see it render at the link I added. https://katapod.com/this-page-does-not-exist
But it infinitely reloads.

**Comment by Aerilym** [comment, marked as answer]
After some investigation, a `router.refresh()` was hidden somewhere. We're using supabase, and the refresh was linked to the authentication resetting. I assume the 404 page triggered an auth reset, and so triggered the router refresh.
